### DISCUSSION TITLE
Backend API ,

### DISCUSSION THREAD

**Discussion by aamir-gmail** [infiniflow/ragflow#2526] [General]
I am running private docker, I want to use the backend API to do just a document similarity search, in an automated manner, once I have loaded the documents in the knowledgebase, because I am doing bulk search and match, using the Web UI would be impractical. 

Any help would be much appreciated.

**Comment by KevinHuSh** [comment]
Please refer to [this](https://github.com/infiniflow/ragflow/blob/main/sdk/python/test/t_document.py#L284)

**Reply by aamir-gmail** [reply]
Thank you,  I have already setup the document store in the knowledge base, I just needed the API for Search, like the one you have got for chat 
![image](https://github.com/user-attachments/assets/4711d694-a2f8-428e-b4fd-64e272484ada)

**Reply by aamir-gmail** [reply]
Where would I find the API key  RAGFlow(API_KEY, HOST_ADDRESS)
def test_parse_document_and_chunk_list(self):
        rag = RAGFlow(API_KEY, HOST_ADDRESS)
        ds = rag.create_dataset(name="God7")
        name='story.txt'
        path = 'test_data/story.txt'
        # name = "Test Document rag.txt"
        # blob = " Sample document content for rag test66. rag wonderful apple os documents apps. Sample document content for rag test66. rag wonderful apple os documents apps.Sample document content for rag test66. rag wonderful apple os documents apps.Sample document content for rag test66. rag wonderful apple os documents apps. Sample document content for rag test66. rag wonderful apple os documents apps. Sample document content for rag test66. rag wonderful apple os documents apps. Sample document content for rag test66. rag wonderful apple os documents apps. Sample document content for rag test66. rag wonderful apple os documents apps. Sample document content for rag test66. rag wonderful apple os documents apps.  Sample document content for rag test66. rag wonderful apple os documents apps. Sample document content for rag test66. rag wonderful apple os documents apps. Sample document content for rag test66. rag wonderful apple os documents apps."
        rag.create_document(ds, name=name, blob=open(path, "rb").read())
        doc = rag.get_document(name=name)
        doc.async_parse()

**Reply by aamir-gmail** [reply]
any specific port  ipaddress:port

**Reply by aamir-gmail** [reply]
can I use this 
API_KEY = 'ragflow-k0YzUxMGY4NjY5YTExZWY5MjI5MDI0Mm'
HOST_ADDRESS = 'http://127.0.0.1:9380'

**Reply by KevinHuSh** [reply]
I think yes.

**Reply by aamir-gmail** [reply]
Thank you again for your prompt reply, what API would I be using to call "SEARCH" , as I mentioned I already have the knowledgebase setup.

**Reply by KevinHuSh** [reply]
Please refer to [this](https://github.com/infiniflow/ragflow/blob/main/sdk/python/test/t_document.py#L284)
